dask/array/utils.py

```diff
@@ -156,8 +156,10 @@ def compute_meta(func, _dtype, *args, **kwargs):
                 else:
                     return None
             except ValueError as e:
-                # min/max functions have no identity, attempt to use the first meta
-                if "zero-size array to reduction operation" in str(e):
+                # min/max functions have no identity, just use the same input type when there's only one
+                if len(
+                    args_meta
+                ) == 1 and "zero-size array to reduction operation" in str(e):
```

Comment:
As noted in https://github.com/dask/dask/issues/7668. @pentschev, curious about your thoughts here. Personally, I don't think we should have this case at all; I don't think it's a safe assumption to make in generality for any user-defined function. But I do see the convenience.

Reply:
I agree with your assessment, t's not a safe assumption for user-defined functions indeed and I would as well prefer not to need this. However, without this condition attempting to compute meta for a 0-D downstream array (e.g., CuPy) will fall into the general `except Exception` case and `return None`, which will cause meta to be a NumPy array, and thus break those downstream use cases. The problem is we have no known alternative, this is why we check the specific exception message, for any other corner cases we would have to extend this function to handle them properly, but user-defined functions should guarantee proper handling of 0-D arrays to avoid such issues.